keras_contrib/layers/capsule.py

```diff
@@ -0,0 +1,177 @@
+       the spatial relationships between objects (in images)
+       or words (in text) by encoding additional information
+       about the image or text, such as angle of rotation,
+       thickness and brightness, relative proportions etc.
+       This layer can be used instead of pooling layers to
+       lower dimensions and still capture important information
+       about the relationships and structures within the data.
+       A normal pooling layer would lose a lot of
+       this information.
+
+       This layer can be used on the output of any layer
+       which has a 3-D output (including batch_size). For example,
+       in image classification, it can be used on the output of a
+       Conv2D layer for Computer Vision applications. Also,
+       it can be used on the output of a GRU or LSTM Layer
+       (Bidirectional or Unidirectional) for NLP applications
+
+       # Example usage :
+           1). COMPUTER VISION
+
+           input_image = Input(shape=(None, None, 3))
+
+           conv_2d = Conv2D(64,
+                            (3, 3),
+                            activation='relu')(input_image)
+
+           capsule = Capsule(num_capsule=10,
+                             dim_capsule=16,
+                             routings=3,
+                             share_weights=True)(conv_2d)
+
+           2). NLP
+
+           maxlen = 72
+           max_features = 120000
+           input_text = Input(shape=(maxlen,))
+
+           embedding = Embedding(max_features,
+                                 embed_size,
+                                 weights=[embedding_matrix],
+                                 trainable=False)(input_text)
+
+           bi_gru = Bidirectional(GRU(64,
+                                      return_seqeunces=True))(embedding)
+
+           capsule = Capsule(num_capsule=5,
+                             dim_capsule=5,
+                             routings=4,
+                             share_weights=True)(bi_gru)
+
+       # Arguments
+           num_capsule : Number of Capsules (int)
+           dim_capsules : Dimensions of the vector output of each Capsule (int)
+           routings : Number of dynamic routings in the Capsule Layer (int)
+           share_weights : Whether to share weights between Capsules or not
+           (boolean)
+
+       # Input shape
+            3D tensor with shape:
+            (batch_size, input_num_capsule, input_dim_capsule)
+            [any 3-D Tensor with the first dimension as batch_size]
+
+       # Output shape
+            3D tensor with shape:
+            (batch_size, num_capsule, dim_capsule)
+
+       # References
+        - [Dynamic-Routing-Between-Capsules]
+          (https://arxiv.org/pdf/1710.09829.pdf)
+        - [Keras-Examples-CIFAR10-CNN-Capsule]"""
+
+    def __init__(self,
+                 num_capsule,
+                 dim_capsule,
+                 routings=3,
+                 share_weights=True,
+                 activation='squash',
```

Comment:
Default activation should be `None` (linear) to be consistent with the rest of keras. You should add in the docstring that a capsule network is mainly used a the `squash` activation. But it should not be possible to use the squash activation with just a string (to be again consistant with keras). Please show an example of how to use the squash activation.

Reply:
Changed the default activation to 'None'.  Added an example of how to use the squash function as activation in the docstring. Removed the option of using the squash activation function with a string.